app/src/main/java/com/example/kuba10/mypokemonplaces/AddPlaceFragment/AddPlaceFragment.java

```diff
@@ -1,169 +0,0 @@
-package com.example.kuba10.mypokemonplaces.AddPlaceFragment;
-
-
-import android.content.Context;
-import android.os.Bundle;
-import android.support.v4.app.Fragment;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.view.ViewGroup;
-import android.widget.Button;
-import android.widget.ImageView;
-import android.widget.TextView;
-
-import com.example.kuba10.mypokemonplaces.ChooseFragment.ChooseFragment;
-import com.example.kuba10.mypokemonplaces.Constants;
-import com.example.kuba10.mypokemonplaces.FragmentListener;
-import com.example.kuba10.mypokemonplaces.Model.PokePlace;
-import com.example.kuba10.mypokemonplaces.Model.PokemonGo;
-import com.example.kuba10.mypokemonplaces.R;
-import com.google.android.gms.maps.model.LatLng;
-import com.squareup.picasso.Picasso;
-
-import java.util.ArrayList;
-import java.util.Date;
-
-import butterknife.BindView;
-import butterknife.ButterKnife;
-
-
-
-public class AddPlaceFragment extends Fragment {
-
-    private FragmentListener fragmentListener;
-    private ArrayList<PokemonGo> pokemonGo_data_list;
-    private PokemonGo selectedPokemon;
-
-    @BindView(R.id.pokemon_image_view)
-    ImageView pokemonImageView;
-    @BindView(R.id.description_input)
-    TextView descriptionField;
-    @BindView(R.id.title_input)
-    TextView titleField;
-    @BindView(R.id.add_button)
-    Button addButt;
-    @BindView(R.id.cancel_button)
-    Button cancelButt;
-
-    public static AddPlaceFragment newInstance(LatLng position) {
-        AddPlaceFragment fragment = new AddPlaceFragment();
-        Bundle args = new Bundle();
-        args.putParcelable(Constants.POSITION, position);
-        fragment.setArguments(args);
-        return fragment;
-    }
-
-
-    @Override
-    public void onAttach(Context context) {
-        super.onAttach(context);
-        fragmentListener = (FragmentListener) context;
```

Comment:
Assertuj, czy aby context na pewno implementuje FragmentListener. `if(context instanceOf FragmentListener)...`

Reply:
czy w tej asercji zostawia się jakiś else? else null albo else: komunikat dla uzytkownika?

Reply:
Ostatecznie rzecz biorac nie ma w tym programie opcji zeby cos innego niz MainActivity wywołało fragment - jest tylko jedna aktywnosc i tylko ona implementuje ten interface

Reply:
Tak jasne, jest to jednak dobra praktyka programistyczna (co jeśli będziesz chciał przerzucić to do innego projektu, albo po 6 miesiącach pojawi się nowe activity itp.).

W else rzucaj wyjątkiem. Zobacz sobie jak zrobili to w przykładzie w Android Studio (zaznacz wszystkie checkboxy generując przykładowy fragment)